simple_uart.c

```
@@ -82,8 +82,11 @@ int simple_uart_read(struct simple_uart *sc, void *buffer, int max_len)
         SetCommTimeouts(sc->port, &commTimeout);
     }
 
-    if (!ReadFile (sc->port, buffer, max_len, (LPDWORD)&r, NULL) != 0)
+    Sleep(50);
```

I'm not sure about this `Sleep(50);` - is that necessary? That will put a pretty hard rate-limit on the serial port, which doesn't seem ideal.

---

Hi André, the Problem on My Site was, that i Misses sporadically Bytes. This wait solved ist for me. But ist should also possible to solve ist on My application

---

If possible, I'd rather that this wasn't in `simple_uart.c` - some applications will want to fail early if no data is immediately available.

---

I Van adjust the pr

---

Yes, could you drop this line from the PR? Thanks

---

removed